### PR TITLE
feat: comment replies ack 922

### PR DESCRIPTION
**Pull request by christian14b**
Clicking `reply` opens a comment input. Then, clicking `comment` will emit the event and close the input

image

### CODE REVIEW

File: @kiva/kv-components/vue/KvCommentsReplyButton.vue [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<!-- eslint-disable max-len -->
+<template>
+	<button
```
Comment (dyersituations) [inline @kiva/kv-components/vue/KvCommentsReplyButton.vue:3]: In this repo we tend to add simple unit tests for new components. This component should get a test to validate the event is emitted.

File: @kiva/kv-components/vue/KvCommentsListItem.vue [hunk 2]
```diff
@@ -99,17 +112,34 @@ export default {
...
-			});
+			if (reaction === REPLY_COMMENT_EVENT) {
+				showInput.value = true;
+				nextTick(() => {
+					commentsAddRef.value.$refs.input.focus();
+				});
+			}
+
+			if (value) {
+				props.handleClick({
```
Comment (dyersituations) [inline @kiva/kv-components/vue/KvCommentsListItem.vue:127]: Not sure why I didn't notice this before, but this should be an emitted event, rather than a prop method.

File: @kiva/kv-components/vue/KvCommentsAdd.vue [hunk 3]
```diff
@@ -87,14 +94,21 @@ export default {
 
 		const comment = () => {
 			emit(ADD_COMMENT_EVENT, addCommentValue.value);
+			if (props.userMention) {
+				emit(HIDE_INPUT_EVENT);
```
Comment (dyersituations) [inline @kiva/kv-components/vue/KvCommentsAdd.vue:98]: Why do we only hide if `userMention`? Is this prop incorrectly named?

File: @kiva/kv-components/vue/KvCommentsListItem.vue (hunk 2 above)
Comment (roger-in-kiva) [reply @kiva/kv-components/vue/KvCommentsListItem.vue:127]: Agreed! I can add this to the components tweaks as building a generic payload...

File: @kiva/kv-components/tests/unit/specs/components/KvCommentsList.spec.js [hunk 4]
```diff
@@ -31,7 +31,7 @@ describe('KvCommentsList', () => {
 		};
 
 		await userEvent.click(replyButton);
-		expect(emitted()[REPLY_COMMENT_EVENT]).toEqual([[{ ...TEST_OBJ, reaction: REPLY_COMMENT_EVENT }]]);
```
Comment (dyersituations) [inline @kiva/kv-components/tests/unit/specs/components/KvCommentsList.spec.js:34]: Why was this line removed?

File: @kiva/kv-components/tests/unit/specs/components/KvCommentsListItem.spec.js [hunk 5]
```diff
@@ -37,7 +37,6 @@ describe('KvCommentsListItem', () => {
 		const replyButton = getByRole('button', { name: 'Reply' });
 
 		await userEvent.click(replyButton);
-
-		expect(handleClick).toHaveBeenCalledTimes(1);
```
Comment (dyersituations) [inline @kiva/kv-components/tests/unit/specs/components/KvCommentsListItem.spec.js:41]: Why was this removed?

File: @kiva/kv-components/vue/KvCommentsListItem.vue (hunk 2 above)
Comment (dyersituations) [reply @kiva/kv-components/vue/KvCommentsListItem.vue:127]: @christian14b @roger-in-kiva Will this be a follow-up task?

Comment (roger-in-kiva) [reply @kiva/kv-components/vue/KvCommentsListItem.vue:127]: Yeah, I have this here: https://kiva.atlassian.net/browse/ACK-1027

File: @kiva/kv-components/vue/KvCommentsAdd.vue (hunk 3 above)
Comment (dyersituations) [reply @kiva/kv-components/vue/KvCommentsAdd.vue:101]: I would probably just rename this prop to something like `isReply`. "User mention" has different potential meaning, for example tagging a user.

File: @kiva/kv-components/tests/unit/specs/components/KvCommentsListItem.spec.js (hunk 5 above)
Comment (dyersituations) [reply @kiva/kv-components/tests/unit/specs/components/KvCommentsListItem.spec.js:41]: @roger-in-kiva FYI since you said it was added to another ticket.

File: @kiva/kv-components/vue/KvCommentsAdd.vue [hunk 6]
```diff
@@ -74,27 +76,38 @@ export default {
...
 
 		const cancel = () => {
 			addCommentValue.value = '';
+			emit(HIDE_INPUT_EVENT);
 		};
 
 		const comment = () => {
 			emit(ADD_COMMENT_EVENT, addCommentValue.value);
+			addCommentValue.value = '';
+			emit(HIDE_INPUT_EVENT);
```
Comment (dyersituations) [inline @kiva/kv-components/vue/KvCommentsAdd.vue:99]: Don't we want to check `isReply` here?

**Review by dyersituations — APPROVED** [review]

**Review by roger-in-kiva — APPROVED** [review]
